lib/ogsf/gp3.c

```
@@ -295,5 +304,9 @@ int Gp_load_sites_thematic(geosite *gp, struct Colors *colors)
             _("%d points without category. "
               "Unable to determine color rules for features without category."),
             nskipped);
+    if (driver) {
+        db_close_database_shutdown_driver(driver);
+        Vect_destroy_field_info(Fi);
```

The call of `Vect_destroy_field_info(Fi);` can be lifted out of the `if (driver)` statement.
(If `Fi` is NULL Vect_destroy_field_info() silently returns, no need to check).

---

done changes